packages/amplify_datastore/example/integration_test/model_type_test.dart

```diff
@@ -0,0 +1,426 @@
+    group('List<int>', () {
+      var list = List.generate(3, (i) => i);
+      var models = List.generate(5, (_) => IntListTypeModel(value: list));
+      testModelOperations<IntListTypeModel>(
+        models: models,
+        classType: IntListTypeModel.classType,
+      );
+    });
+
+    group(
+      'List<int> (with null value)',
+      () {
+        var models = List.generate(5, (_) => IntListTypeModel());
+        testModelOperations<IntListTypeModel>(
+          models: models,
+          classType: IntListTypeModel.classType,
+        );
+      },
+      // unskip when https://github.com/aws-amplify/amplify-flutter/issues/813 is resolved
+      skip: true,
+    );
+
+    group('double', () {
+      var models =
+          List.generate(5, (i) => DoubleTypeModel(value: i.toDouble()));
+      testModelOperations<DoubleTypeModel>(
+        models: models,
+        classType: DoubleTypeModel.classType,
+      );
+    });
+
+    group('double (with null value)', () {
+      var models = List.generate(5, (i) => DoubleTypeModel());
+      testModelOperations<DoubleTypeModel>(
+        models: models,
+        classType: DoubleTypeModel.classType,
+      );
+    });
+
+    group('List<double>', () {
+      var list = List.generate(3, (i) => i.toDouble());
+      var models = List.generate(5, (_) => DoubleListTypeModel(value: list));
+      testModelOperations<DoubleListTypeModel>(
+        models: models,
+        classType: DoubleListTypeModel.classType,
+      );
+    });
+
+    group('List<double> (with null value)', () {
+      var models = List.generate(5, (_) => DoubleListTypeModel());
+      testModelOperations<DoubleListTypeModel>(
+        models: models,
+        classType: DoubleListTypeModel.classType,
+      );
+    });
+
+    group('bool', () {
+      var models =
+          List.generate(5, (i) => BoolTypeModel(value: i == 0 ? false : true));
+      testModelOperations<BoolTypeModel>(
+        models: models,
+        classType: BoolTypeModel.classType,
+      );
+    });
+
+    group('bool (with null value)', () {
+      var models = List.generate(5, (i) => BoolTypeModel());
+      testModelOperations<BoolTypeModel>(
+        models: models,
+        classType: BoolTypeModel.classType,
+      );
+    });
+
+    group('List<bool>', () {
+      var list = List.generate(3, (i) => i == 0 ? false : true);
+      var models = List.generate(5, (_) => BoolListTypeModel(value: list));
+      testModelOperations<BoolListTypeModel>(
+        models: models,
+        classType: BoolListTypeModel.classType,
+      );
+    });
+
+    group('List<bool> (with null value)', () {
+      var models = List.generate(5, (_) => BoolListTypeModel());
+      testModelOperations<BoolListTypeModel>(
+        models: models,
+        classType: BoolListTypeModel.classType,
+      );
+    });
+
+    group('TemporalDate', () {
+      var now = DateTime.now();
+      var models =
+          List.generate(5, (i) => DateTypeModel(value: TemporalDate(now)));
```

Comment:
Good op maybe to throw in some edge cases like:

```dart
TemporalDate.fromString('1970-01-01T00:00:00Z')
TemporalDate.fromString('2199-12-31T23:59:59Z')
```

Same for other temporal fields